### PR TITLE
Create empty parsable changelog if .versionbot folder is present

### PR DESCRIPTION
**Pull request by nazrhom**
Change-type: minor
Signed-off-by: Giovanni Garufi <giovanni@balena.io>

### CODE REVIEW

File: tests/cli/cases/parsable-changelog.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  {
+    commits: [
+      {
+        subject: 'Implement x',
+        body: '',
+        footer: {
+          'Changelog-Entry': 'Implement x',
+          'changelog-entry': 'Implement x',
+          'Change-Type': 'minor',
+          'change-type': 'minor'
```
Comment (lurch) [inline tests/cli/cases/parsable-changelog.js]: Given that this is both machine-generated and machine-parsable, perhaps there's no point in including both capitalisation options i.e. both `Changelog-Entry` and `changelog-entry` ?

File: tests/cli/cases/parsable-changelog.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+const parsableChangelog = _.map(
+  yaml.safeLoad(shelljs.cat('.versionbot/CHANGELOG.yml').stdout),
+  (entry) => {
+    entry.commits = _.map(entry.commits, (commit) => {
+      return _.omit(commit, 'hash');
+    });
+    return _.omit(entry, 'date');
+  });
+console.log(JSON.stringify(parsableChangelog));
```
Comment (lurch) [inline tests/cli/cases/parsable-changelog.js]: Left-over debug code?
(I wonder if `console.log` is something that one of the ResinCI checks could look for? :man_shrugging: )

**Comment by lurch** [comment]
I guess this indirectly closes #45 ?

File: lib/versionist.js [hunk 3]
```diff
@@ -248,9 +249,17 @@ exports.generateChangelog = (commits, options = {}, callback) => {
...
-      const changelog = yaml.safeLoad(fs.readFileSync(parsableChangelogPath, 'utf8'));
+    const parsableChangelogFolder = '.versionbot';
+    const parsableChangelogName = 'CHANGELOG.yml';
+    const parsableChangelogPath = path.join(parsableChangelogFolder, parsableChangelogName);
+    if (fs.existsSync(parsableChangelogFolder)) {
+      let changelog = [];
+      try {
+        changelog = yaml.safeLoad(fs.readFileSync(parsableChangelogPath, 'utf8'));
+      } catch (e) {
+        // If no parsableChangelog exists, assume its empty
```
Comment (hedss) [inline lib/versionist.js]: Might well be worth catching on a `ENOENT` here, just to ensure it doesn't exist, rather then there being some other sort of fileop/error happening.

File: tests/cli/cases/parsable-changelog.js (hunk 2 above)
Comment (hedss) [reply tests/cli/cases/parsable-changelog.js]: Sometimes you want it though. Maybe we should have an explicit test logging lib. :)

File: tests/presets.spec.js [hunk 4]
```diff
@@ -925,6 +925,7 @@ describe('Presets', function() {
 
         it('should be able to update the version in both files', function(done) {
           presets.updateVersion.cargo({}, this.cwd.name, '1.0.1', (error) => {
+
```
Comment (hedss) [inline tests/presets.spec.js]: Strange extra empty line has crept in?

**Pull request closed without merging**